File: src/main/java/top/gerritchang/tools/dao/QueryListDao.java
package top.gerritchang.tools.dao;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Map;

@Component
public class QueryListDao {

    @Autowired
    private JdbcTemplate jdbcTemplate;

    public Map getSqlByScriptName(String scriptName) {
        return jdbcTemplate.queryForMap("select script_sql,script_method from form_script where script_name = '"
                + scriptName + "'");
    }

    public List<Map> query(String script_sql) {
        return jdbcTemplate.queryForList(script_sql, Map.class);
    }
}

File: src/main/java/top/gerritchang/tools/dao/QueryListDao.kt
package top.gerritchang.tools.dao

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Component

@Component
class QueryListDao {

    @Autowired
    private lateinit var jdbcTemplate:JdbcTemplate

    fun getSqlByScriptName(scriptName: String): Map<String, *> {
        return jdbcTemplate.queryForMap("select script_sql,script_method from form_script where script_name = '"
                + scriptName + "'")
    }

    fun query(script_sql: String): List<Map<*, *>?> {
        return jdbcTemplate.queryForList(script_sql, MutableMap::class.java)
    }

}
